dev/src/reference.ts

```diff
@@ -2832,6 +2854,285 @@ export class CollectionReference<T = firestore.DocumentData>
+    return this._query;
+  }
+
+  get aggregates(): T {
+    return this._aggregates;
+  }
+
+  get(): Promise<AggregateQuerySnapshot<T>> {
+    return this._get();
+  }
+
+  /**
+   * Internal get() method that accepts an optional transaction id.
+   *
+   * @private
+   * @internal
+   * @param {bytes=} transactionId A transaction ID.
+   */
+  _get(transactionId?: Uint8Array): Promise<AggregateQuerySnapshot<T>> {
+    // Capture the error stack to preserve stack tracing across async calls.
+    const stack = Error().stack!;
+
+    return new Promise((resolve, reject) => {
+      const stream = this._stream(transactionId);
+      stream.on('error', err => {
+        reject(wrapError(err, stack));
+      });
+      stream.once('data', result => {
+        stream.destroy();
+        resolve(result);
+      });
+      stream.on('end', () => {
+        reject('No AggregateQuery results');
+      });
+    });
+  }
+
+  /**
+   * Internal streaming method that accepts an optional transaction ID.
+   *
+   * @param transactionId A transaction ID.
+   * @private
+   * @internal
+   * @returns A stream of document results.
+   */
+  _stream(transactionId?: Uint8Array): Readable {
+    const tag = requestTag();
+    const firestore = this._query.firestore;
+
+    const stream: Transform = new Transform({
+      objectMode: true,
+      transform: (proto: api.IRunAggregationQueryResponse, enc, callback) => {
+        if (proto.result) {
+          const readTime = Timestamp.fromProto(proto.readTime!);
+          const data = this.decodeResult(proto.result);
+          callback(
+            undefined,
+            new AggregateQuerySnapshot<T>(this, readTime, data)
+          );
+        } else {
+          callback(Error('unexpected'));
+        }
+      },
+    });
+
+    firestore
+      .initializeIfNeeded(tag)
+      .then(async () => {
+        // `toProto()` might throw an exception. We rely on the behavior of an
+        // async function to convert this exception into the rejected Promise we
+        // catch below.
+        const request = this.toProto(transactionId);
+
+        let streamActive: Deferred<boolean>;
+        do {
+          streamActive = new Deferred<boolean>();
+          const backendStream = await firestore.requestStream(
+            'runAggregationQuery',
+            /* bidirectional= */ false,
+            request,
+            tag
+          );
+          stream.on('close', () => {
+            backendStream.resume();
+            backendStream.end();
+          });
+          backendStream.on('error', err => {
+            backendStream.unpipe(stream);
+            // If a non-transactional query failed, attempt to restart.
+            // Transactional queries are retried via the transaction runner.
+            if (!transactionId && !isPermanentRpcError(err, 'runQuery')) {
+              logger(
+                'Query._stream',
+                tag,
+                'Query failed with retryable stream error:',
```

Comment:
Do you need to put a space after the ':' character? Also on line 2983 below.

Reply:
This is copied code from the Query class _stream method.

I investigated, and under the covers it uses the `util.format` function that formats correctly.